invenio_rdm_records/requests/access/requests.py

```diff
@@ -59,6 +62,14 @@ class GuestSubmitAction(actions.SubmitAction):
     def execute(self, identity, uow):
         """Execute the submit action."""
         self.request["title"] = self.request.topic.resolve().metadata["title"]
+
+        record = RDMRecord.pid.resolve(self.request["topic"]["record"])
+        exp = record.parent.access.settings.secret_link_expiration
+        expires_at = (
+            arrow.now().shift(days=exp).date().isoformat() if exp is not None else ""
```

Comment:
It feels like that the `secret_link_expiration` field should be None/null when never expires, instead of empty strings.
This would avoid many `if == ''` or `if != ''` around. Marshmallow loading/serializing should control this.

Ping @max-moser: any opinion? is it a big change?